src/pydex_app/orderbook.py

```diff
@@ -3,30 +3,133 @@
+        asset_data_a -- string asset_data for maker side (i.e. `maker_asset_data`)
+        asset_data_b -- string asset_data for taker side (i.e. `taker_asset_data`)
+        """
+        asset_pairs = []
+        normalized_asset_a = normalize_query_param(asset_data_a)
+        normalized_asset_b = normalize_query_param(asset_data_b)
+        if asset_data_a and asset_data_b:
+            asset_pairs = SignedOrder.query.with_entities(
+                SignedOrder.maker_asset_data, SignedOrder.taker_asset_data
+            ).filter(
+                (SignedOrder.order_status > 0)
+                & (SignedOrder.maker_asset_data == normalized_asset_a)
+                & (SignedOrder.taker_asset_data == normalized_asset_b)
+            )
+        elif asset_data_a:
+            asset_pairs = SignedOrder.query.with_entities(
+                SignedOrder.maker_asset_data, SignedOrder.taker_asset_data
+            ).filter(
+                (SignedOrder.order_status > 0)
+                & (SignedOrder.maker_asset_data == normalized_asset_a)
+            )
+        elif asset_data_b:
+            asset_pairs = SignedOrder.query.with_entities(
+                SignedOrder.maker_asset_data, SignedOrder.taker_asset_data
+            ).filter(
+                (SignedOrder.order_status > 0)
+                & (SignedOrder.maker_asset_data == normalized_asset_b)
+            )
+        else:
+            asset_pairs = SignedOrder.query.with_entities(SignedOrder.maker_asset_data, SignedOrder.taker_asset_data)
+        unique_asset_pairs = set(asset_pairs)
+
+        def erc721_asset_data_to_asset(asset_data):
+            return {
+                "minAmount": ZERO_STR,
+                "maxAmount": "1",
+                "precision": 0,
+                "assetData": asset_data
+            }
+
+        def erc20_asset_data_to_asset(asset_data):
+            return {
+                "minAmount": ZERO_STR,
+                "maxAmount": MAX_INT_STR,
+                "precision": DEFAULT_ERC20_DECIMALS,
+                "assetData": asset_data
+            }
+
+        def asset_data_to_asset(asset_data):
+            asset_proxy_id: str = asset_data[:SELECTOR_LENGTH]
+            if asset_proxy_id == abi_utils.method_id("ERC20Token", ["address"]):
+                return erc20_asset_data_to_asset(asset_data)
+            if asset_proxy_id == abi_utils.method_id("ERC721Token", ["address"]):
+                return erc721_asset_data_to_asset(asset_data)
+            raise ValueError(f"Invalid asset data {str(asset_data)}")
+
+        def get_asset_pair_data(asset_pair):
+            return {
+                "assetDataA": asset_data_to_asset(asset_pair[0]),
+                "assetDataB": asset_data_to_asset(asset_pair[1])
+            }
+
+        asset_pairs_data = [get_asset_pair_data(asset_pair) for asset_pair in list(unique_asset_pairs)]
+        return paginate(asset_pairs_data, page=page, per_page=per_page), len(asset_pairs_data)
```

Comment:
I think you should sor the asset_pairs_data before paginating because the order is not guaranteed from database pull and you want to make sure at least as long as the live orders is not changing the unique set of pairs, the pagination will be predictable. Also, generally, for our own thing, we will probably whitelist the pairs in a database and just retrieve those rather than getting the pairs on the fly...